experiments/mountain_car_plots.py

```diff
@@ -0,0 +1,269 @@
+    env.env.state = ob
+    next_ob, *_ = env.step(act)
+    next_obs.append(next_ob)
+  return np.array(next_obs)
+
+
+def make_policy_rollout(policy_path, env_name="MountainCar-v0",
+                        ) -> List[util.rollout.Trajectory]:
+  """Load policy from path and return a list of Trajectories."""
+  venv = DummyVecEnv([lambda: gym.make(env_name)])
+  with load_policy("ppo2", str(policy_path), venv) as gen_policy:
+    trajs = util.rollout.generate_trajectories(
+      gen_policy, venv, sample_until=util.rollout.min_episodes(50))
+  return list(trajs)
+
+
+def make_heatmap(
+    act: int,
+    reward_fn: RewardFn,
+    pos_step_size: float = 0.1,
+    velocity_step_size: float = 1e-2,
+    mark_goal: bool = True,
+    gen_trajs: Optional[List[util.rollout.Trajectory]] = None,
+    exp_trajs: Optional[List[util.rollout.Trajectory]] = None,
+    legend_on: bool = True,
+    title: bool = None,
+    heatmap: bool = True,
+    filter_trans_by_act: bool = True,
+) -> None:
+  """
+  Make a MountainCar heatmap of rewards for a particular action. X axis
+  is position. Y axis is velocity.
+
+  Args:
+    act: The MountainCar action number whose reward we are evaluating.
+      Should be 0, 1, or 2.
+    reward_fn: Reward function. Should accept unnormalized inputs.
+    pos_step_size: The x axis granularity.
+    velocity_step_size: The y axis granularity.
+    mark_goal: If True then plot a line for goal position.
+    gen_trajs: A list of generator trajectories to
+      scatterplot on top of the heatmap.
+    exp_trajs: A list of exp trajectories to scatterplot on
+      top of the heatmap.
+    legend_on: Whether to plot the legend.
+    title: Custom title.
+    heatmap: Whether to plot the heatmap.
+    filter_trans_by_act: If True, then filter out transitions from
+      `gen_trajs` and `exp_trajs` that don't use action `act` before
+      scatterplotting.
+  """
+  assert 0 <= act <= 2
+
+  obs_vec, acts_vec = [], []
+  for p in np.arange(MC_POS_MIN, MC_POS_MAX + 1e-5, step=pos_step_size):
+    for v in np.arange(MC_VEL_MIN, MC_VEL_MAX + 1e-5, step=velocity_step_size):
+      obs_vec.append([p, v])
+      acts_vec.append(act)
+
+  obs_vec = np.array(obs_vec)
+  acts_vec = np.array(acts_vec)
+  next_obs_vec = make_next_obs(obs_vec, acts_vec)
+
+  R = reward_fn(obs_vec, acts_vec, next_obs_vec, None)
+
+  df = pd.DataFrame()
+  df["position"] = np.round(obs_vec[:, 0], 5)
+  df["velocity"] = np.round(obs_vec[:, 1], 5)
+  df["reward"] = R
+  df = df.pivot("velocity", "position", "reward")
+
+  def convert_to_rc(pos, vel):
+    x = (np.array(pos) - MC_POS_MIN) / pos_step_size + 0.5
+    y = (np.array(vel) - MC_VEL_MIN) / velocity_step_size + 0.5
+    return x, y
+
+  def convert_to_rc_filtered(trajs):
+    trans = util.rollout.flatten_trajectories(trajs)
+    obs = trans.obs
+    if filter_trans_by_act:
+      obs = obs[trans.acts == act]
+    pos, vel = obs[:, 0], obs[:, 1]
+    return convert_to_rc(pos, vel)
+
+  plt.figure()
```

Comment:
```suggestion
fig, ax = plt.subplots()
```
Less error prone to explicitly pass axis around. Most important if you're working with multiple figures.